core/src/fields.rs

```diff
@@ -542,23 +550,66 @@ impl MongoFields {
+
+                        let current_col_metadata_response: ResultSetSchema = if mongo_connection
+                            .cluster_type
+                            == MongoClusterType::AtlasDataFederation
+                        {
+                            let get_schema_cmd = doc! {"sqlGetSchema": &collection_name};
+
+                            let sql_get_schema_response: Result<SqlGetSchemaResponse> =
+                                mongodb::bson::from_document(
+                                    db.run_command(get_schema_cmd).await.unwrap(),
+                                )
+                                .map_err(|e| {
+                                    Error::CollectionDeserialization(collection_name.clone(), e)
+                                });
+
+                            if let Err(error) = sql_get_schema_response {
+                                // If there is an Error while deserializing the schema, we won't show any columns for it
+                                warnings.push(error);
+                                continue;
+                            }
+
+                            sql_get_schema_response.unwrap().into()
+                        } else if mongo_connection.cluster_type == MongoClusterType::Enterprise {
+                            let schema_collection =
+                                db.collection::<Document>(SQL_SCHEMAS_COLLECTION);
+
+                            let schema_doc: Document = mongo_connection
+                                .runtime
+                                .block_on(async {
+                                    schema_collection
+                                        .find_one(doc! {
+                                            "_id": &collection_name
+                                        })
+                                        .await
+                                        .map_err(Error::QueryExecutionFailed)
+                                })?
+                                .ok_or(Error::SchemaDocumentNotFoundInSchemaCollection(vec![
+                                    collection_name.clone(),
+                                ]))?;
+
+                            let result_set_schema: Result<ResultSetSchema> =
+                                ResultSetSchema::from_sql_schemas_document(&schema_doc).map_err(
+                                    |e| {
+                                        Error::CollectionDeserialization(collection_name.clone(), e)
+                                    },
+                                );
+
+                            if let Err(error) = result_set_schema {
+                                // If there is an Error while deserializing the schema, we won't show any columns for it
+                                warnings.push(error);
+                                continue;
+                            }
+
+                            result_set_schema.unwrap()
```

Comment:
How about something like this to avoid the unwrap, it has similar behavior:
```
match result_set_schema {
    Ok(schema) => schema,
    Err(error) => {
        warnings.push(error);
        continue;
    }
}
```